source/bao_hyp/config.rst

```diff
@@ -65,39 +65,468 @@ describe the image running on the VM (see `Guest Image`_);
+physical mappings. This feature provides a means to define the physical memory region explicitly.
 
 3. Inter-Process Communication (IPC)
 ####################################
+.. _Inter-Process Communication (IPC):
+
+Inter-Process Communication (IPC) enables seamless communication between distinct processes in a
+computing system. This mechanism facilitates data exchange, synchronization, and coordination,
+fostering enhanced functionality and system efficiency.
+
+.. figure:: img/ipc.svg
+    :align: center
+    :width: 100%
+    :name: ipc-fig
+
+|
+
+Bao provides support for IPC, allowing VMs to establish communication channels. The IPC
+configuration involves defining the number of IPCs using the ``ipc_num`` field within the
+``vm_config`` struct. The specifics of each IPC are then outlined through the ``ipcs`` structure,
+including fields such as ``base``, ``size``, ``shmem_id``, ``interrupt_num``, and ``interrupts``.
+
+- ``ipc_num`` [optional] - defines the number of IPCs assigned to the VM. By default, ``ipc_num``
+  equals to zero;
+- ``ipcs`` [mandatory if ``ipc_num`` > 0] - corresponds to the specification of the IPC and is
+  configured through the following structure:
+
+.. code-block:: c
+
+    struct ipc {
+        paddr_t base;
+        size_t size;
+        size_t shmem_id;
+        size_t interrupt_num;
+        irqid_t *interrupts;
+    };
+
+
+where:
+
+- ``base``  [mandatory] - corresponds to the virtual base address of the IPC memory region;
+- ``size`` [mandatory] - corresponds to the size of the IPC memory region;
+
+.. note::
+  The ``size`` field must be less than or equal to the size of the shared memory. Additionally, for
+  MPU systems, the ``base`` field is ignored, as the region address is the same as the shared
+  memory object address. Also, it is mandatory for both ``base`` and ``size`` to be aligned with
+  the architecture's smallest page size. For MMU systems, this corresponds to 4K in all
+  architectures, while for MPU systems, the alignment corresponds to 64 bytes.
+
+- ``shmem_id`` [mandatory] - corresponds to the ID of the shared memory associated with the IPC;
+- ``interrupt_num`` [mandatory] - defines the number of interrupts assigned to the IPC;
+- ``interrupts`` [mandatory if *interrupt_num* > 0] - defines a list of interrupt IDs assigned to
+  the IPC - ``(irqid_t[]) {irq_1, ..., irq_n}``;
+
+.. warning::
+  Specifying a number of interrupts in the ``interrupts`` buffer that differs from the
+  ``interrupt_num`` may result in unforeseen behavior.
 
 4. Devices
 ##########
+.. _Devices:
+
+- ``dev_num`` [mandatory] - corresponds to the number of devices assigned to the VM;
+- ``devs`` [mandatory if *dev_num* > 0] - corresponds to the specification of the VM's devices and
+  is configured through the following structure:
+
+.. code-block:: c
+
+    struct vm_dev_region {
+        paddr_t pa;
+        vaddr_t va;
```

Comment:
Currently device configuration is awkward for MPU configuration as there is no va

I suggest to state this clearly

We should work on this at some point

Reply:
Thanks! Tried to address this point by updating the note in https://github.com/bao-project/bao-docs/pull/78/commits/3558ec91451e0f20fb210ad2178de2abc561258f, but I'm not sure if it fully addresses this point.